cli/lsp/language_server.rs

```diff
@@ -1069,8 +1069,7 @@ impl Inner {
 
   fn walk_workspace(config: &Config) -> (BTreeSet<ModuleSpecifier>, bool) {
     let mut workspace_files = Default::default();
-    let document_preload_limit =
-      config.workspace_settings().document_preload_limit;
+    let entry_limit = 1000;
```

Comment:
I don't fully understand this, so I'll have to look at it in the morning, but I'm a bit concerned that a user would run into the same bug again once they have a lot of files.

Reply:
They'd need 1000 files at the top level preceding `deno.json` for that to happen. After the multi-deno.json detection is all set up, if there's demand I want to make some of it more lazy, like scanning upwards for new scopes whenever a file is opened in the editor. Let's leave that for later though.